RP-Core/src/main/java/abilitiesV2/skills/abitilies/knowledge/gateofknowledge/GateOfKnowledge.java

```diff
@@ -0,0 +1,73 @@
+import core.entityTypes.WorldEntity;
+import me.libraryaddict.disguise.disguisetypes.Disguise;
+import org.bukkit.Location;
+import org.bukkit.entity.EntityType;
+
+import java.util.List;
+
+public class GateOfKnowledge extends Skill {
+
+    private static final int DURATION_SECONDS = 30;
+    private static final String DESCRIPTION = String.format("You summon & open the Gate of Knowledge, a legendary structure " +
+                    "that is said to grant wishes of its summoners. Once the gate opens, the gate's representative will descend in " +
+                    "order to hear out your wish. Legend has it that there is only a number of wishes in the world's history that have " +
+                    "been witnessed to be granted, but the infomation about the exact words required to formulate them is incomplete or " +
+                    "yet to be discovered. The gate lasts %s seconds. \n" +
+                    "There is no guarantee that the wish will be granted in a way that satisfies the summoner, or that it will be granted " +
+                    "at all. It should be noted that the consequences of summoning the gate may be dangerous to inexperienced summoners.",
+            DURATION_SECONDS
+    );
+    private static final Tier TIER = Tier.V;
+    private static final String TRIGGER_WORD = "wisnaris";
+    private static final TriggerType TRIGGER_TYPE = TriggerType.VOCAL;
+
+    @Override
+    public BaseSkillName getBaseSkillName() {
+        return BaseSkillName.GATE_OF_KNOWLEDGE;
+    }
+
+    @Override
+    public SkillDefinition getBaseSkillDefinition() {
+        return new SkillDefinitionBuilder()
+                .name(getBaseSkillName())
+                .description(DESCRIPTION)
+                .tier(TIER)
+                .castRequirements()
+                .addTrigger(TRIGGER_TYPE, new Trigger(TRIGGER_WORD))
+                .build();
+    }
+
+    @Override
+    protected void invoke(WorldEntity caster, List<String> args) {
+        Location gateLocation = caster.getLocation().get().clone(); // TODO once schematic
+        Location spawnLocation = gateLocation.clone().add(gateLocation.getDirection().multiply(2));
+        Disguise disguise = Appearance.asMob(EntityType.EVOKER);
+        EmissaryPersonality emissaryPersonality = new StandardPersonality(disguise);
+        KnowledgeEmissary knowledgeEmissary = new KnowledgeEmissary(spawnLocation, emissaryPersonality);
+
+        VariableEffect hasGate = new VariableEffect.Builder()
+                .type(VariableEffect.Type.HAS_GATE_OF_KNOWLEDGE)
+                .duration(DURATION_SECONDS)
+                .applyActions(() -> {
+                    knowledgeEmissary.speak(EmissaryMessageType.WELCOME);
+                })
+                .clearActions(() -> {
+                    knowledgeEmissary.speak(EmissaryMessageType.GOODBYE);
+                })
+                .build();
+        hasGate.apply(caster);
+    }
```

Comment:
Nitpick, but why have a variable?
```suggestion
        VariableEffect hasGate = new VariableEffect.Builder()
                .type(VariableEffect.Type.HAS_GATE_OF_KNOWLEDGE)
                .duration(DURATION_SECONDS)
                .applyActions(() -> {
                    knowledgeEmissary.speak(EmissaryMessageType.WELCOME);
                })
                .clearActions(() -> {
                    knowledgeEmissary.speak(EmissaryMessageType.GOODBYE);
                })
                .build()
                .apply(caster);
    }
```